app/models/changeset.rb

```diff
@@ -84,11 +70,33 @@ def error
 
   private
 
+  def find_comparison
+    if empty?
+      NullComparison.new('Start and end commits are the same')
+    else
+      Rails.cache.fetch(cache_key) do
+        GITHUB.compare(repo, previous_commit, commit)
+      end
+    end
+  rescue Octokit::NotFound, Octokit::InternalServerError => e
+    error_msg = case e
+      when Octokit::NotFound then "Commit not found"
+      when Octokit::InternalServerError then "Internal error #{e.message}"
+      else
+        "Unknown error"
```

Comment:
if you indent `case` then you get this weirdness at the end that always looks like something is out of sync

```
  else
    "xxx"
end
```

vs 

```
else
  "xxx"
end
```